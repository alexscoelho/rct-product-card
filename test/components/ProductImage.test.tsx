import React from 'react';
import renderer from 'react-test-renderer';

import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('show component with image from context', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('show component with image from props', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://placekitten.com/640/360" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
