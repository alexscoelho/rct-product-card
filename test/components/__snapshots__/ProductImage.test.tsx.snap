// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage show component with image from context 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage show component with image from props 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://placekitten.com/640/360"
/>
`;
